template.tpl

```diff
@@ -1680,6 +1680,205 @@ ___WEB_PERMISSIONS___
+  code: |-
+    const excludeReferrers = "test.com";
+    const expectedConfig = {
+      defaultTracking: {
+        attribution: {
+          excludeReferrersText: [excludeReferrers],
+          resetSessionOnNewCampaign: undefined,
+          initialEmptyValue: 'EMPTY',
+        },
+        pageViews: false,
+        sessions: false,
+        fileDownloads: false,
+        formInteractions: false,
+      }
+    };
+
+    mockData.type = 'init';
+    mockData.defaultEventTracking = true;
+    mockData.detAttribution = true;
+    mockData.attributionExcludeReferrers = excludeReferrers;
+
+    // Test the following line:
+    // _amplitude(instanceName, 'init', data.apiKey, initUserId, generateConfiguration());
+    mock('copyFromWindow', key => {
+      return function() {
+        assertThat(arguments[0], 'Incorrect instance name').isEqualTo(mockData.instanceName);
+        assertThat(arguments[1], 'Incorrect tag type').isEqualTo(mockData.type);
+        assertThat(arguments[2], 'Incorrect apiKey object').isEqualTo(mockData.apiKey);
+        assertThat(arguments[3], 'Incorrect user Id').isEqualTo(null);
+        assertThat(arguments[4], 'Incorrect config').isEqualTo(expectedConfig);
+
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Init tag with multiple exclude referrers
+  code: |-
+    const excludeReferrers = ["test.com", "text.org"];
+    const expectedConfig = {
+      defaultTracking: {
+        attribution: {
+          excludeReferrersText: excludeReferrers,
+          resetSessionOnNewCampaign: undefined,
+          initialEmptyValue: 'EMPTY',
+        },
+        pageViews: false,
+        sessions: false,
+        fileDownloads: false,
+        formInteractions: false,
+      }
+    };
+
+    mockData.type = 'init';
+    mockData.defaultEventTracking = true;
+    mockData.detAttribution = true;
+    mockData.attributionExcludeReferrers = excludeReferrers;
+
+    // Test the following line:
+    // _amplitude(instanceName, 'init', data.apiKey, initUserId, generateConfiguration());
+    mock('copyFromWindow', key => {
+      return function() {
+        assertThat(arguments[0], 'Incorrect instance name').isEqualTo(mockData.instanceName);
+        assertThat(arguments[1], 'Incorrect tag type').isEqualTo(mockData.type);
+        assertThat(arguments[2], 'Incorrect apiKey object').isEqualTo(mockData.apiKey);
+        assertThat(arguments[3], 'Incorrect user Id').isEqualTo(null);
+         assertThat(arguments[4], 'Incorrect config').isEqualTo(expectedConfig);
+      };
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Init tag with single exclude referrers regex
+  code: |-
+    const excludeReferrersRegex = ["test.com", "text.org"];
```

Comment:
This should be a single exclude referrers instead an array?

Reply:
Nice catch! Updated